### ISSUE TITLE
(mintty) enabling 'cursorline' only highlights text, not the whole screen line

### ISSUE THREAD

**Issue by rex-ludorum** [neovim/neovim#25609]
### Problem

In my .vimrc, I have

```
set cursorline
hi clear CursorLine
hi CursorLine gui=underline cterm=underline
```

to enable underlining. In Vim, this underlines the whole line in the screen as expected, even without whitespace trailing the text:
![msys_vim](https://github.com/neovim/neovim/assets/22417446/c586bc5e-d7cb-42b3-9f97-c0a4d9895030)

However, in Neovim (my nvim setup sources my .vimrc), this only underlines the portion of the line with text along with leading whitespace:
![msys_neovim1](https://github.com/neovim/neovim/assets/22417446/251df4be-8e85-4410-bc16-e082a171ff19)

Rarely, it doesn't underline leading indentations or manages to underline the rest of the line. I have no clue what the criteria for these cases are:
![msys_neovim3](https://github.com/neovim/neovim/assets/22417446/78519596-0216-4373-87a7-a43a5d180bf4)
![msys_neovim2](https://github.com/neovim/neovim/assets/22417446/d40e36e4-fa73-49b9-b771-0e03929419ae)

Moreover, this is only a problem on Windows, where I have nvim installed on msys through pacman. On Ubuntu with the same config, the whole line is always underlined:
![ubuntu-neovim](https://github.com/neovim/neovim/assets/22417446/f66bdc87-b9a3-4247-8688-9f490fd15daa)

I'd like to have neovim on Windows behave consistently with Ubuntu and underline the whole line.

### Steps to reproduce

See above

### Expected behavior

The whole line should be underlined, not only the portion with text.

### Neovim version (nvim -v)

0.9.2

### Vim (not Nvim) behaves the same?

no, 9.0.1976

### Operating system/version

Windows 10

### Terminal name/version

mintty 3.6.5

### $TERM environment variable

xterm-256color

### Installation

pacman on msys2
`pacman -S mingw-w64-ucrt-x86_64-neovim`

**Comment by justinmk** [comment]
Latest stable release is 0.9.4, not 0.9.2. Also try the prerelease https://github.com/neovim/neovim/releases/tag/nightly .

> Terminal name/version : mintty 3.6.5

Does it happen if with cmd.exe or Windows terminal ?

**Comment by rex-ludorum** [comment]
I updated to 0.9.4 and the problem persists. Running nvim in cmd or Windows Terminal fixes it though.

**Comment by rex-ludorum** [comment]
I tried installing my setup on a different Windows 11 computer and the problem is fixed on there. The problem only happens on my Windows 10 computer.

**Comment by justinmk** [comment]
Then this is likely a terminal issue and not going to to spend time adding a workaround.

**Comment by mintty** [comment]
nvim was logged to send wrong escape sequences in the documented scenario:
https://github.com/mintty/mintty/issues/1239#issuecomment-1774199317

**Comment by zeertzjq** [comment]
@rex-ludorum Do you have `infocmp` command available in MSYS2? Try searching and installing some packages containing `ncurses` in their names if it's not present. If `infocmp` is available, what does `infocmp -Lx` print in mintty?

**Comment by mintty** [comment]
infocmp does not depend on the terminal, it only considers $TERM
I've tried to vary on xterm-256color etc already.

**Comment by zeertzjq** [comment]
Yes, but I don't get `\x1b[16X` or `\x1b[m` in XTerm. What I get is 16 spaces and `\x1b[0;4m`.

**Comment by rex-ludorum** [comment]
I do have it installed. $infocmp -V$ gives `ncurses 6.4.20221231`.

`infocmp -Lx` gives [infocmp-out.txt](https://github.com/neovim/neovim/files/13072467/infocmp-out.txt)

**Comment by zeertzjq** [comment]
Hmm, that's also the same as what I get in XTerm on Linux, so it's really strange that these sequences are quite different from what I get in XTerm.

**Comment by mintty** [comment]
@justinmk, this is not a terminal issue as has been demonstrated in https://github.com/mintty/mintty/issues/1239#issuecomment-1774199317.
It might be a bug induced by the conhost layer of Windows 10, or it might be wrong usage of escape sequences by neovim (see referred comment), which I cannot check easily.

**Comment by zeertzjq** [comment]
This is a ConPTY bug and it's already fixed in ConPTY: https://github.com/microsoft/terminal/issues/13229

**Comment by zhangweize9-cyber** [comment]
You'll need to contact the author who developed this theme.
